### ISSUE TITLE
Reference treated as [AMP]

### ISSUE THREAD

**Issue by kintel** [uncrustify/uncrustify#2031]
This seems to be a regression introduced by https://github.com/uncrustify/uncrustify/pull/1616

`uncrustify -p debug.txt -c test.cfg -f test.cpp`

```
#include <string>

void evaluate(std::string &filename, class MyClass *myptr);
```

```
sp_after_byref=remove
sp_after_byref_func=remove
```

...results in `void evaluate(std::string & filename, class MyClass *myptr);`

-> the space after `&` was not removed.

debug.txt
```
# Uncrustify-0.67-485-09a5be76
sp_after_byref                  = remove
sp_after_byref_func             = remove
# option(s) with 'not default' value: 2
#
# -=====-
# Line                Tag              Parent          Columns Br/Lvl/pp     Flag   Nl  Text
#   1>            PREPROC[         PP_INCLUDE][  1/  1/  2/  0][0/0/0][ 100080001][0-0] #
#   1>         PP_INCLUDE[               NONE][  2/  2/  9/  0][0/0/0][         1][0-0]  include
#   1>             STRING[               NONE][ 10/ 10/ 18/  1][0/0/0][         1][0-0]          <string>
#   1>            NEWLINE[               NONE][ 18/ 18/  1/  0][0/0/0][         0][2-0]
#   3>               TYPE[               NONE][  1/  1/  5/  0][0/0/0][     70000][0-0] void
#   3>      FUNC_CTOR_VAR[               NONE][  6/  6/ 14/  1][0/0/0][   1800000][0-0]      evaluate
#   3>        FPAREN_OPEN[      FUNC_CTOR_VAR][ 14/ 14/ 15/  0][0/0/0][ 100000000][0-0]              (
#   3>               TYPE[               NONE][ 15/ 15/ 18/  0][0/1/0][     40008][0-0]               std
#   3>          DC_MEMBER[               NONE][ 18/ 18/ 20/  0][0/1/0][ 100000008][0-0]                  ::
#   3>               TYPE[               NONE][ 20/ 20/ 26/  0][0/1/0][         8][0-0]                    string
#   3>                AMP[               NONE][ 27/ 27/ 28/  1][0/1/0][ 100000008][0-0]                           &
#   3>               WORD[               NONE][ 29/ 28/ 36/  0][0/1/0][         8][0-0]                             filename
#   3>              COMMA[               NONE][ 37/ 36/ 37/  0][0/1/0][ 100000008][0-0]                                     ,
#   3>              CLASS[               NONE][ 39/ 38/ 43/  1][0/1/0][     50008][0-0]                                       class
#   3>               WORD[               NONE][ 45/ 44/ 51/  1][0/1/0][         8][0-0]                                             MyClass
#   3>              ARITH[               NONE][ 53/ 52/ 53/  1][0/1/0][ 100000008][0-0]                                                     *
#   3>               WORD[               NONE][ 54/ 53/ 58/  0][0/1/0][     40008][0-0]                                                      myptr
#   3>       FPAREN_CLOSE[      FUNC_CTOR_VAR][ 59/ 58/ 59/  0][0/0/0][ 100000000][0-0]                                                           )
#   3>          SEMICOLON[      FUNC_CTOR_VAR][ 60/ 59/ 60/  0][0/0/0][ 100000000][0-0]                                                            ;
#   3>            NEWLINE[               NONE][ 61/ 60/  1/  0][0/0/0][         0][1-0]
# -=====-
```

**Comment by guy-maurel** [comment]
The problem seams to be with the part
`..., class MyClass ...`
Can you push a more complete, compileable input file?

**Comment by kintel** [comment]
As `MyClass` is just a forward-declaration, this should be a minimally compilable example.

**Comment by guy-maurel** [comment]
I ask once more:
Can you push a more complete, compileable input file?

**Comment by andrey-starodubtsev** [comment]
1. Minimal compilable piece is

```c++
#include <string>

void evaluate(std::string &filename, class MyClass *myptr);

int main() {
    return 0;
}
```

2. This issue was fixed in 2340884 and can be closed.

**Comment by kintel** [comment]
Thx!
@guy-maurel Sorry, I didn't realize that you also wanted the example to be linkable.
